src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -548,13 +553,14 @@ private static Long findSchema(Connection connection, BinlogPosition targetPosit
 			PreparedStatement s = connection.prepareStatement(
 				"SELECT id from `schemas` "
 				+ "WHERE deleted = 0 "
-				+ "AND ((binlog_file < ?) OR (binlog_file = ? and binlog_position <= ?)) AND server_id = ? "
-				+ "ORDER BY binlog_file DESC, binlog_position DESC limit 1");
+				+ "AND last_heartbeat_read <= ? AND ((binlog_file < ?) OR (binlog_file = ? and binlog_position <= ?)) AND server_id = ? "
+				+ "ORDER BY last_heartbeat_read DESC, binlog_file DESC, binlog_position DESC limit 1");
```

Comment:
Should we just order by `last_heartbeat_read DESC, id DESC` since binlog_file is not indicative of ordering if DB is restored from backup?